### PR TITLE
feat: Add gatekeeper chart

### PR DESCRIPTION
**Pull request by jimmidyson**
This will be pushed upstream ASAP but we would like
to include it in Mesosphere charts repo quickly and
demo it for customers.

### CODE REVIEW

**Comment by jimmidyson** [comment]
The failure is a bit rubbish:

> staging/gatekeeper/values.yaml
> [17:30:52][Step 1/1]   77:1      error    too many blank lines (1 > 0)  (empty-lines)

Blank lines are useful to make the file more readable and is consistent with upstream charts.

File: staging/gatekeeper/values.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+# To _fail closed_ on failures, change to Fail. During initial testing, we
+# recommend leaving the failure policy as Ignore.
+admissionControllerFailurePolicy: Ignore
+
+# Adds a namespace selector to the admission controller webhook
+admissionControllerNamespaceSelector:
+  matchExpressions:
+  - {key: validation.gatekeeper.sh/disable-validation, operator: NotIn, values: ["true"]}
+
```
[the remark this answers is not in the document]
Comment (jimmidyson) [reply staging/gatekeeper/values.yaml]: Haha a blank line at the end of the file?! :man_facepalming: I'll remove it.

[the remark this answers is not in the document]
Comment (jimmidyson) [reply staging/gatekeeper/values.yaml]: Ah double blank lines at end of file... Wow. OK removed.

File: staging/gatekeeper/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        {{- if .Values.securityContext.runAsUser }}
+        runAsUser: {{ .Values.securityContext.runAsUser }}
+        {{- end }}
+        {{- if .Values.securityContext.fsGroup }}
+        fsGroup: {{ .Values.securityContext.fsGroup }}
+        {{- end }}
+      {{- end }}
+      containers:
+      - args:
+        - --auditInterval=30
```
Comment (hectorj2f) [inline staging/gatekeeper/templates/deployment.yaml]: what about allowing to configure this value ?

**Comment by hectorj2f** [comment]
@jimmidyson is this PR ready ?

**Comment by alejandroEsc** [comment]
In the future, please do not add to charts that create CRDS without first handling them correctly.

**Comment by jimmidyson** [comment]
@alejandroEsc I'm pretty sure that this chart was added before the current CRD handling pattern was employed.

**Pull request merged**